src/SSVOpenHexagon/Core/Discord.cpp

```diff
@@ -89,32 +89,75 @@ bool discord_manager::set_rich_presence_in_menu()
     }
 
     discord::Activity activity{};
-    activity.SetState("In menu");
-    activity.SetDetails("...");
+    activity.SetState("Selecting Level");
+    activity.SetDetails("");
+    discord::ActivityTimestamps& currentTimestamp = activity.GetTimestamps();
+    currentTimestamp.SetStart(static_cast<int64_t>(
+        std::chrono::high_resolution_clock::now().time_since_epoch().count() /
+        1000000000));
+    _core->ActivityManager().UpdateActivity(activity, [](discord::Result r) {
+        if(r != discord::Result::Ok)
+        {
+            ssvu::lo("Discord") << "Fail\n";
+        }
+    });
+
+    return true;
+}
+
+bool discord_manager::set_rich_presence_on_replay()
+{
+    if(!_initialized)
+    {
+        return false;
+    }
+
+    discord::Activity activity{};
+    activity.SetState("Watching Replay");
+    activity.SetDetails("");
+    discord::ActivityTimestamps& currentTimestamp = activity.GetTimestamps();
+    currentTimestamp.SetStart(0);
     _core->ActivityManager().UpdateActivity(activity, [](discord::Result r) {
         if(r != discord::Result::Ok)
         {
-            ssvu::lo("Discord") << "FAil\n";
+            ssvu::lo("Discord") << "Fail\n";
         }
     });
 
     return true;
 }
 
-bool discord_manager::set_rich_presence_in_game(const std::string& str_status)
+bool discord_manager::set_rich_presence_in_game(
+    const std::string& level_info, const std::string& second_info, bool dead)
 {
     if(!_initialized)
     {
         return false;
     }
 
     discord::Activity activity{};
-    activity.SetState("In game");
-    activity.SetDetails(str_status.data());
+    activity.SetDetails(("Playing " + level_info).data());

```

Comment:
Is this well-defined behavior? You're creating a string temporary, giving its pointer to `SetDetails`, and then it dies afterwards. If `SetDetails` copies the characters then it's fine, but it just keeps a reference/pointer then it's UB. We should make sure and add a comment in the code if it's safe.

Reply:
Here is what `SetDetails` does:
```c++
void Activity::SetDetails(char const* details)
{
    strncpy(internal_.details, details, 128);
    internal_.details[128 - 1] = '\0';
}
```
So it just copies the string over. And a comment should be added in the code to clarify.

Reply:
I'm also wondering if he wants me to remove "Playing " which I am completely fine with. It's not necessary information.